data_spec_validator/spec/defines.py

```diff
@@ -165,7 +162,33 @@ class CheckerOP(Enum):
 
 
 class Checker:
-    def __init__(self, checks, op=CheckerOP.ALL, extra=None):
-        self.checks = checks
-        self.op = op
+    def __init__(self, checks, optional=False, op=CheckerOP.ALL, extra=None):
+        """
+        checks: list of str(Check)
+        optional: boolean
+                  Set optional to True, the validation process will be passed if the field is absent
+        op: CheckerOP
+        extra: None or Dict
+        """
+        self.checks = checks or []
+        self._op = op
+        self._optional = optional
         self.extra = extra or {}
+
+        self._ensure()
+
+    def _ensure(self):
+        if self._optional and len(self.checks) == 0:
```

Comment:
nit: `if self._optional and not self.checks:`

PEP-8: For sequences, (strings, lists, tuples), use the fact that empty sequences are false

Reply:
NICE! Good to know that, but sorry I merged it as quick as a flash